internal/dag/gatewayapi_processor.go

```diff
@@ -221,24 +221,35 @@ func (p *GatewayAPIProcessor) getListenerRouteKinds(listener gatewayapi_v1alpha2
 
 	var routeKinds []gatewayapi_v1alpha2.Kind
 
-	// TODO if a kind is invalid, set "ResolvedRefs: false" condition
-	// on the listener with "InvalidRouteKinds" reason.
-
 	for _, routeKind := range listener.AllowedRoutes.Kinds {
-		if routeKind.Group == nil {
-			p.Errorf("Listener.AllowedRoutes.Group not specified")
-			continue
-		}
-		if *routeKind.Group != gatewayapi_v1alpha2.GroupName {
-			p.Errorf("Listener.AllowedRoutes.Group %q not supported", *routeKind.Group)
+		if routeKind.Group != nil && *routeKind.Group != gatewayapi_v1alpha2.GroupName {
+			gwAccessor.AddListenerCondition(
+				string(listener.Name),
+				gatewayapi_v1alpha2.ListenerConditionResolvedRefs,
+				metav1.ConditionFalse,
+				gatewayapi_v1alpha2.ListenerReasonInvalidRouteKinds,
+				fmt.Sprintf("Group %q is not supported", *routeKind.Group),
+			)
 			continue
 		}
 		if routeKind.Kind != gatewayapi_v1alpha2.Kind(KindHTTPRoute) && routeKind.Kind != gatewayapi_v1alpha2.Kind(KindTLSRoute) {
```

Comment:
nit: Could get rid of the `gatewayapi_v1alpha2.Kind()` and just compare directly. 

e.g.
```
if routeKind.Kind != KindHTTPRoute && routeKind.Kind != KindTLSRoute {
}
``